src/main/java/uk/ac/ebi/eva/pipeline/io/writers/AnnotationMongoWriter.java

```diff
@@ -0,0 +1,192 @@
+ * <p>
+ * 20_63963_G/A    20:63963   A  ENSG00000178591    ENST00000382410    Transcript upstream_gene_variant  -  -  -  -  -  -  DISTANCE=4388;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=protein_coding;CANONICAL=YES;CCDS=CCDS12989.2;ENSP=ENSP00000371847;SWISSPROT=DB125_HUMAN;TREMBL=B2R4E8_HUMAN;UNIPARC=UPI00001A36DE
+ * 20_63963_G/A    20:63963   A  ENSG00000178591    ENST00000608838    Transcript upstream_gene_variant  -  -  -  -  -  -  DISTANCE=3928;STRAND=1;SYMBOL=DEFB125;SYMBOL_SOURCE=HGNC;HGNC_ID=18105;BIOTYPE=processed_transcript
+ * <p>
+ * Will be:
+ * <p>
+ * "annot" : {
+ * "ct" : [
+ * { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000382410", "codon" : "-", "strand" : "+", "bt" : "protein_coding", "aaChange" : "-", "so" : [ 1631 ] },
+ * { "gn" : "DEFB125", "ensg" : "ENSG00000178591", "enst" : "ENST00000608838", "codon" : "-", "strand" : "+", "bt" : "processed_transcript", "aaChange" : "-",
+ * "so" : [ 1631 ] } ],
+ * "xrefs" : [
+ * { "id" : "DEFB125", "src" : "HGNC" },
+ * { "id" : "ENST00000382410", "src" : "ensemblTranscript" },
+ * { "id" : "ENST00000608838", "src" : "ensemblTranscript" },
+ * { "id" : "ENSG00000178591", "src" : "ensemblGene"
+ */
+public class AnnotationMongoWriter extends MongoItemWriter<Annotation> {
+    private static final Logger logger = LoggerFactory.getLogger(AnnotationMongoWriter.class);
+
+    private static final String ANNOTATION_XREF_ID_FIELD = "xrefs.id";
+
+    private static final String ANNOTATION_CT_SO_FIELD = "ct.so";
+
+    private MongoOperations mongoOperations;
+
+    private String collection;
+
+    private String vepVersion;
+
+    private String vepCacheVersion;
+
+    public AnnotationMongoWriter(MongoOperations mongoOperations,
+                                 String collection,
+                                 String vepVersion,
+                                 String vepCacheVersion) {
+        super();
+        Assert.notNull(mongoOperations, "A Mongo instance is required");
+        Assert.hasText(collection, "A collection name is required");
+
+        setCollection(collection);
+        setTemplate(mongoOperations);
+
+        this.mongoOperations = mongoOperations;
+        this.collection = collection;
+        this.vepVersion = vepVersion;
+        this.vepCacheVersion = vepCacheVersion;
+
+        createIndexes();
+    }
+
+    @Override
+    protected void doWrite(List<? extends Annotation> annotations) {
+        Map<String, List<Annotation>> annotationsByStorageId = groupAnnotationById(annotations);
+
+        for (Map.Entry<String, List<Annotation>> annotationsIdEntry : annotationsByStorageId.entrySet()) {
+            String storageId = annotationsIdEntry.getKey();
+            List<Annotation> annotationsById = annotationsIdEntry.getValue();
+
+            Annotation annotation = annotationsById.get(0);
+
+            if (annotationsById.size() > 1) {
+                annotation = concatenateOtherAnnotations(
```

Comment:
given that we are grouping the annotation by id in both writers, what do you think about changing the reader so that it only outputs one annotation for all the lines that talk about the same variant?